vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java

```diff
@@ -225,42 +273,84 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
    * @param query  query used to match documents
    * @param resultHandler will be called when complete
    */
-  @Fluent
+  @Deprecated @Fluent
   MongoClient remove(String collection, JsonObject query, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Remove matching documents from a collection and return the handler with MongoClientDeleteResult result
+   *
+   * @param collection  the collection
+   * @param query  query used to match documents
+   * @param resultHandler will be called when complete
+   */
+  @Fluent
+  MongoClient removeWithMongoClientDeleteResult(String collection, JsonObject query, Handler<AsyncResult<MongoClientDeleteResult>> resultHandler);
+
+  /**
    * Remove matching documents from a collection with the specified write option
    *
    * @param collection  the collection
    * @param query  query used to match documents
    * @param writeOption  the write option to use
    * @param resultHandler will be called when complete
    */
-  @Fluent
+  @Deprecated @Fluent
   MongoClient removeWithOptions(String collection, JsonObject query, WriteOption writeOption, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Remove matching documents from a collection with the specified write option and return the handler with MongoClientDeleteResult result
+   *
+   * @param collection  the collection
+   * @param query  query used to match documents
+   * @param writeOption  the write option to use
+   * @param resultHandler will be called when complete
+   */
+  @Fluent
+  MongoClient removeWithOptionsWithMongoClientDeleteResult(String collection, JsonObject query, WriteOption writeOption, Handler<AsyncResult<MongoClientDeleteResult>> resultHandler);
+
+  /**
    * Remove a single matching document from a collection
    *
    * @param collection  the collection
    * @param query  query used to match document
    * @param resultHandler will be called when complete
    */
-  @Fluent
+  @Deprecated @Fluent
   MongoClient removeOne(String collection, JsonObject query, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Remove a single matching document from a collection and return the handler with MongoClientDeleteResult result
+   *
+   * @param collection  the collection
+   * @param query  query used to match document
+   * @param resultHandler will be called when complete
+   */
+  @Fluent
+  MongoClient removeOneWithMongoClientDeleteResult(String collection, JsonObject query, Handler<AsyncResult<MongoClientDeleteResult>> resultHandler);
```

Comment:
suggested naming `removeDocument` - not actually sure why we have this as a separate method from removeDocuments though...